### ISSUE TITLE
[BUG] serverless deploy fail through the nx plugin 

### ISSUE THREAD

**Issue by dbrrt** [flowaccount/nx-plugins#55]
**Issue**
During the deployment the serverless plugin stops at packaging the service

**To Reproduce**

1. Configure your AWS ID and secret
2. generate a serverless app with the cli `nx g @flowaccount/serverless:app ...`
3. execute deploy target: `nx deploy myapp`

**Expected behavior**
The CLI should pass the packaging step, it simply stops with no error.

**Screenshots**

**Check which provider is affected:**
[X] AWS

**Check which framework is affected:**
[x] Serverless

**Additional context**
The generated app requires sls below 2.0, current version is 2.6.0
The tsconfig file extended is looking at the wrong path '../../tsconfig.json' instead of  '../../tsconfig.base.json'

OS: Linux 5.4.0-48-generic x86_64
Distribution: Linux Mint

Might be related to
https://github.com/flowaccount/nx-plugins/issues/50

**Comment by wickstargazer** [comment]
Is this for angular universal app?

**Comment by dbrrt** [comment]
@wickstargazer It's for serverless framework plugin.

**Comment by Destreyf** [comment]
I just wanted to chime in here.

I am using a self hosted drone.io and cannot get the build to succeed, however if i run it on my laptop it fails every other time it's run.

What's even stranger is even with debug set to true in my env.json and setting SLS_DEBUG to true, i get inconsistent debug messages.

To improve my odds of deployment i have broken my builds into 2 steps.

```sh
# First we create a package, I modified my angular.json to have this run target.
npx ng run api:package
# Next i run the deploy on the built asset
SLS_DEBUG=true npx serverless deploy --package dist/apps/api/.serverless/ --config apps/api/serverless.yml
```

The line in my angular.json file is as follows
```json
        "package": {
          "builder": "@flowaccount/nx-serverless:sls",
          "options": {
            "command": "package",
            "buildTarget": "api:build:production",
            "config": "apps/api/serverless.prod.yml",
            "location": "dist/apps/api",
            "package": "dist/apps/api"
          }
        },
```
This is placed just below the "serve" option in my config for the api "project".

I had to install YARN for the sole purpose of even getting the builds to work due to the dependency tree shaking that this tries to do.

I really like this project, I just wish the builds were more reliable, instead of having to run the package command 2-4 times before I can run a deploy.

I have an other project I plan on using nx-serverless on, I'm planning on spending more time looking into the cause of the build issues as it prevents me from being able to do an automated deploy from my CI/CD system.

**Comment by Destreyf** [comment]
To further clarify, I do not have issues with SLS_DEBUG when doing other serverless tasks, only tasks related to nx-serverless.

**Comment by dbrrt** [comment]
@Destreyf Same, with nx-serverless, It's stopping for me at `packaging` step, with no error, where using sls commands standalone, the deployment processes just fine.

**Comment by Destreyf** [comment]
@dbrrt Can you add the build config to your angular.json file and try running the following command:
```bash
SLS_DEBUG=true npx ng run api:package
```

Does it show any messages?  I do get past packaging, albeit inconsistently.

**Comment by dbrrt** [comment]
@Destreyf I'm not using angular.json, will try in the package.json, are you able to package the lambda with the plugin?

**Comment by Destreyf** [comment]
You may have a workspace.json file that's similar to the angular.json file, basically what I added as a run target for the nx cli that does the packaging.

**Comment by dbrrt** [comment]
Makes sense, I'll try that.

**Comment by wickstargazer** [comment]
Hi i think the above problem is broken down into two issues?
 1. Being the sls documentation is not clear 
 2. The npm doesn't work.
 
Am i right? @Destreyf  I am fixing the npm and will add some documentations you suggested into the readme somewhere.

Thankz

**Comment by Destreyf** [comment]
@wickstargazer I don't know if there's been releases since my last message, the project I used this on has been on hold, I am actually circling back to it this week.

At the time of my message, the real issue was the deploy portion of the SLS command wasn't working consistently, this was actually an issue with the Package step, so I could run the package command multiple times until it succeeded (I had to have yarn installed for it to even work at all) and then once it succeeded because I broke the deploy into two steps, I could successfully deploy.

This was/is particularly frustrating if you're trying to use a CI/CD system, because it always fails on the first attempt no matter what, which CI/CD systems are blank slates each run, essentially making it impossible to automate deployments.

**Comment by wickstargazer** [comment]
@Destreyf i have been facing similar issues after scaling. So i am in fix for that, but i am going to release v0.5.3 in a few days and it will fix the `npm` issue. You will be able to use npm. For the packaging issue there might be several causes which i am investigating. But it might be release as `v1.0.0` in a month time.
